src/utils/Validate.ts

```diff
@@ -0,0 +1,99 @@
+import {inputValid} from '../types/utilsTypes';
+
+export class Validate {
+  private static long = (
+    value: string = '',
+    min: number = 1,
+    max: number = value.length
+  ) => {
+    const length = value.length;
+
+    if (length < min) return 'toShort';
+    if (length > max) return 'toLong';
+  };
+
+  static loginCheck: inputValid = (value) => {
+    const longTest = this.long(value, 3, 20);
+    if (longTest === 'toShort') return 'Слишком короткий логин';
+    if (longTest === 'toLong') return 'Слишком длинный логин';
+
+    const letterTest = value.match(/[a-z]/i);
+    if (!letterTest) return 'В логине должна быть хотя бы одна буква';
+
+    const regTest = value.match(/^[\w-]{3,20}$/);
+    if (!regTest) return 'Недопустимые символы';
+
+    return '';
+  };
```

Comment:
Функция-валидатор должна принимать в себя значение и проверять является ли оно верным или нет. Соответственно она должна возвращаться либо true либо false, и больше ничего.
Представь, что продакт сказал, что текст ошибки нужно изменить или вообще вставить туда ссылку. Эти изменения никак не должны затрагивать логику валидации. Попробуй вынести текст ошибок на уровень компонентов, а в функциях-валидаторах возвращать только boolean. Так код будет менее связным
```suggestion
  static loginCheck = (value: string): boolean => {
    const longTest = this.long(value, 3, 20);
    if (longTest === 'toShort') return false;
    if (longTest === 'toLong') return false;

    const letterTest = value.match(/[a-z]/i);
    if (!letterTest) return false;

    const regTest = value.match(/^[\w-]{3,20}$/);
    if (!regTest) return false;

    return true;
  };
```

Reply:
UPD. Понял, почему ты было сделано так. Ты хотел учесть разные типы ошибок. В таком случае я бы сделал вот так: 

```
type TLoginCheckErros =
  | 'TOO_SHORT'
  | 'TOO_LONG'
  | 'AT_LEAST_ONE_LETTER'
  | 'UNEXPECTED_SYMBOLS'

type Valid = 'VALID';

  static loginCheck = (value: string): TLoginCheckErros | Valid => {
    const longTest = this.long(value, 3, 20);
    if (longTest === 'toShort') return 'TOO_SHORT';
    if (longTest === 'toLong') return 'TOO_LONG';

    const letterTest = value.match(/[a-z]/i);
    if (!letterTest) return 'AT_LEAST_ONE_LETTER';

    const regTest = value.match(/^[\w-]{3,20}$/);
    if (!regTest) return 'UNEXPECTED_SYMBOLS';

    return 'VALID';
  };
```

Reply:
Тут я получше типизировал функцию и обрати внимание на сигнатуру. Прочитав сигнатуру можно четко понять что она делает.
Проверка логика, принимает строку, возвращает либо ошибки, либо инф. о том, что строка валидна. Постарайся так же декларативно оформить остальные валидаторы